Code/autopkglib/__init__.py

```diff
@@ -248,6 +265,7 @@ def _write_json_file(self):
     def _write_plist_file(self):
         """Write out the prefs into a Plist."""
         try:
+            assert self.file_path is not None
```

Comment:
Do you want to leave these assertions in? I feel like this should be surfaced as an exception with a human-readable error message rather than an assert failure traceback.

Reply:
I think they should be left in. Any instance of this assertion tripping is a bug in autopkg, not anything that should be possible during normal usage. You'll note that both `_write_plist_file` and `_write_json_file` are only called through `write_file` which returns if `self.file_path` is `None`. So, the only way this happens if someone calls one of these directly (which nothing does), or the code is changed in such a way that validation goes missing. I'll add an explanation here to the assert like `assert self.file_path is not None, "Internal logic error in preference saving. Please report this to autopkg developers."` to add a little clarity. Though, it's worth noting that this won't cause autopkg to crash, since `AssertionError` derives from `Exception` which both functions swallow anyway...